src/timesync/timesyncd.c

```diff
@@ -68,13 +68,21 @@ static int load_clock_timestamp(uid_t uid, gid_t gid) {
         ct = now(CLOCK_REALTIME);
         if (ct < min) {
                 struct timespec ts;
+                struct tm* tm;
                 char date[FORMAT_TIMESTAMP_MAX];
 
                 log_info("System clock time unset or jumped backwards, restoring from recorded timestamp: %s",
                          format_timestamp(date, sizeof(date), min));
 
                 if (clock_settime(CLOCK_REALTIME, timespec_store(&ts, min)) < 0)
                         log_error_errno(errno, "Failed to restore system clock: %m");
+
+                if (local_rtc)
+                        tm = localtime(&ts.tv_sec);
+                else
+                        tm = gmtime(&ts.tv_sec);
+
+                clock_set_hwclock(tm);
```

Comment:
oh, btw, I#d prefer if we would not do any syncs if we are in local-time mode here... If people run their RTCs in local time, then they should live with the consequences...